include/linux/amlogic/amports/video.h

```diff
@@ -27,12 +27,13 @@ int RDMA_SET_READ(u32 adr);
 
 #define AMVIDEO_MAGIC  'X'
 
-#define AMVIDEO_EXT_GET_CURRENT_VIDEOFRAME _IOR((AMVIDEO_MAGIC), 0x01, int)
+#define AMVIDEO_EXT_GET_CURRENT_VIDEOFRAME _IOR((AMVIDEO_MAGIC), 0x01, u32)
```

Comment:
Do we need compats for these ioctls?
Are you testing with armv7 build of Hyperion?

Reply:
Well my builds are native, using my installer on the Vero4k (so armv8-a).  My CFLAGS are `-march=armv8-a+crc -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard`.  But it's using the 32bit userland `gcc` compiler you supply in `build-essential`.  **I believe it's tantamount to armv7** but feel free to educate me.

Interestingly it wasn't necessary for me to supply any of that for the Hyperion "classic" builds.  But for "nextgen" I found it solved a floating point exception at run-time.

**I don't expect a compatibility layer would be needed** as `u32` is in use throughout the same compilation unit.  When I was merging this macro in I saw that it's original introduction on OSMC with `int` predated OpenELEC by over a year.  But as it was a move towards consistency and it was the only one not be `u32` - [Sam shudders] this amateur didn't see the harm.

Reply:
Can you please build without passing any of this -m optimisations?
(as a test)

Sam

Reply:
>I don't expect a compatibility layer would be needed

I think you'll be alright..

Reply:
Many builds later...

As I mentioned with ng there's a runtime "Floating point exception", which I fixed by exporting the compiler options.  I admit I just used what I had previously for emulators etc, and I wonder now if there's any benefit to doing the same for those as I've uncovered here.
Anyway without passing _at least some_ `-m` flags I can't avoid the runtime error.

Going for the native armv8 I must specify `-march=armv8-a+crc` and/or `-mcpu=cortex-a53`.

If I build for armv7 as is your preference it's `-mcpu=cortex-a7`, like an RPi2.
Using a vague `-march=armv7` or `armv7-a` wont do it, even if I add the relevant floating point elements `-mfpu=neon-vfpv4 -mfloat-abi=hard`.

I'm going with `-mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard` for now then as an armv7 build and will test through the evening for any glaring issues.

Reply:
Great -- let me know how it goes.

Reply:
Just checking in.
A week on and I've not provoked any sync issues or anything else out of the ordinary.

Reply:
Great -- thanks for confirming this.